File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.rs
impl Solution {
    pub fn longest_consecutive(nums: Vec<i32>) -> i32 {
        let uniques: HashSet<&i32> = nums.iter().collect();
        let mut max_length: i32 = 0;
        for num in &nums {
            let mut n: i32 = *num;
            if !uniques.contains(&(n - 1)) {
                let mut curr_length: i32 = 1;
                while uniques.contains(&(n + 1)) {
                    curr_length += 1;
                    n += 1;
                }
                max_length = std::cmp::max(max_length, curr_length);
            } 
        }
        max_length
    }
}

File: Data Structures & Algorithms/house-robber/submission-2.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        if n < 2:
            return nums[0]

        for i in range(1, n):
            m1 = nums[i-1]
            m2 = 0 if i - 2 < 0 else nums[i-2]
            nums[i] = max(nums[i] + m2, m1)
        
        return nums[-1]